icu4c/source/i18n/unitsdata.cpp

```diff
@@ -359,32 +380,20 @@ U_I18N_API UnitPreferences::UnitPreferences(UErrorCode &status) {
     ures_getAllItemsWithFallback(unitsBundle.getAlias(), "unitPreferenceData", sink, status);
 }
 
+// TODO: make outPreferences const?
+//
+// TODO: consider replacing `UnitPreference **&outPrefrences` with slice class
+// of some kind.
 void U_I18N_API UnitPreferences::getPreferencesFor(const char *category, const char *usage,
                                                    const char *region,
-                                                   MaybeStackVector<UnitPreference> *outPreferences,
-                                                   UErrorCode &status) {
-    // UnitPreferenceMetadata *m = getMetadata(category, usage, region);
+                                                   const UnitPreference **&outPreferences,
```

Comment:
You want a non-const pointer into an array of const pointers to const objects.  I'm not 100% sure, but you might need another `const` in this like, like `const UnitPreference* const* &outPreferences`.  The `const` resolution rules have always confused me a little.

Reply:
Done.

(Compiler says I've now got roughly the right combination of asterisks and "const"s. :-)